src/storage-client/src/controller.rs

```diff
@@ -145,7 +145,19 @@ pub enum DataSource {
     /// This source's data is does not need to be managed by the storage
     /// controller, e.g. it's a materialized view, table, or subsource.
     // TODO? Add a means to track some data sources' GlobalIds.
-    Other,
+    Other(DataSourceOther),
+}
+
+/// Describes how data is written to a collection maintained outside of the
+/// storage controller.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum DataSourceOther {
+    /// A user or system table.
+    Table,
+    /// A `MATERIALIZED VIEW` maintained by compute.
+    MaterializedView,
+    /// A subsource.
+    Subsource,
```

Comment:
```suggestion
#[derive(Clone, Debug, Eq, PartialEq)]
pub enum DataSourceOther {
    /// `environmentd` appends timestamped data, i.e. it is a `TABLE`
    TableWrites,
    /// Compute maintains, i.e. it is a `MATERIALIZED VIEW`
    Compute,
    /// Some other sources writes this data, i.e. a subsource.
    Source,
```

Reply:
I think I personally find the new names less clear, but happy to defer to your area expertise. done!